src/async_impl/client.rs

```diff
@@ -2759,7 +2822,8 @@ impl PendingRequest {
                     .body(body)
                     .expect("valid request parts");
                 *req.headers_mut() = self.headers.clone();
-                ResponseFuture::Default(self.client.hyper.request(req))
+                let mut hyper = self.client.hyper.lock().unwrap();
```

Comment:
An alternative to using a Mutex is if the client can be cheaply and safely cloned, then the clone can be mutable.

Reply:
Good idea.